gms/gossiper.cc

```diff
@@ -802,6 +757,84 @@ future<gossiper::endpoint_permit> gossiper::lock_endpoint(inet_address ep) {
+            co_await _messaging.send_gossip_echo(netw::msg_addr(node), generation, max_duration);
+            logger.debug("failure_detector_loop: Send echo to node {}, status = ok", node);
+        } catch (...) {
+            failed = true;
+            logger.info("failure_detector_loop: Send echo to node {}, status = failed: {}", node, std::current_exception());
+        }
+        auto now = gossiper::clk::now();
+        diff = now - last;
+        if (!failed) {
+            last = now;
+        }
+        if (diff > max_duration) {
+            logger.info("failure_detector_loop: Mark node {} as DOWN", node);
+            co_await container().invoke_on(0, [node] (gms::gossiper& g) {
+                return seastar::async([node, &g] {
+                    g.convict(node);
+                });
+            });
+            co_return;
+        } else if (_live_endpoints.size() != nr) {
+            logger.debug("failure_detector_loop: Finish loop for node {}, live_endpoints={}", node, _live_endpoints);
+            co_return;
+        } else  {
+            co_await sleep_abortable(std::chrono::milliseconds(2000), _abort_source);
+        }
+    }
+    co_return;
+}
+
+future<> gossiper::failure_detector_loop() {
+    auto shard = this_shard_id();
+    logger.info("failure_detector_loop: Started on shard={}", shard);
+    while (is_enabled()) {
+        try {
+            utils::chunked_vector<inet_address> nodes;
+            while ((_live_endpoints.size() <= shard) && is_enabled()) {
+                logger.info("failure_detector_loop: Wait for live nodes to work with on shard={}, live_nodes={}, live_nodes_for_this_shard={}",
+                        shard,  _live_endpoints, nodes);
+                co_await sleep_abortable(std::chrono::milliseconds(1000), _abort_source);
+            }
+            nodes = co_await container().invoke_on(0, [shard] (gms::gossiper& g) {
+                utils::chunked_vector<inet_address> nodes;
+                for (unsigned idx = 0; idx < g._live_endpoints.size(); idx++) {
+                    if (idx % smp::count == shard) {
+                        nodes.push_back(g._live_endpoints[idx]);
+                    }
+                }
+                return nodes;
+            });
+            if (nodes.empty()) {
+                logger.debug("failure_detector_loop: Got empty live nodes to work on shard={}, live_nodes={}, live_nodes_for_this_shard={}",
+                        shard, _live_endpoints, nodes);
+                co_await sleep_abortable(std::chrono::milliseconds(1000), _abort_source);
+                continue;
+            }
+            logger.info("failure_detector_loop: Start new round on shard={}, live_nodes={}, live_nodes_for_this_shard={}",
+                    shard, _live_endpoints, nodes);
+            co_await parallel_for_each(nodes, [this] (gms::inet_address& node) {
+                return failure_detector_loop_for_node(node);
+            });
+        } catch (...) {
```

Comment:
Blanket exception ignoring makes debugging unexpected errors harder. 

Do you want to ignore only a specific exception?

Reply:
I should have logged the exception. I did that in v3.  I do not want to ignore any specific exception. If there is error, we should just retry it.